services/thermal_mgr/thermal_mgr.c

```diff
@@ -38,23 +42,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
   thermalMgrQueueHandle = xQueueCreateStatic(
     THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
     thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if (event->type == OS_HANDLER_EVENT){
+    xQueueSendToFront(thermalMgrQueueHandle, event, ( TickType_t ) 0);
+  }  
+  else if (event->type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){
+    xQueueSend(thermalMgrQueueHandle, event, ( TickType_t ) 10 );
+  }
   return ERR_CODE_SUCCESS;
 }
 
+BOOL readTempState = FALSE;
+
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  readTempState = TRUE;
+  thermal_mgr_event_t osHandlerEvent;
+  osHandlerEvent.type = OS_HANDLER_EVENT;
+  thermalMgrSendEvent(&osHandlerEvent);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
+  thermal_mgr_event_type_t queueBuffer;
   while (1) {
-    
+    if( xQueueReceive(thermalMgrQueueHandle, &(queueBuffer),( TickType_t ) 10 ) == pdPASS ){
+      float currentTemp = 0.0;
+      readTempLM75BD(LM75BD_OBC_I2C_ADDR, &currentTemp); 
+      if (readTempState== TRUE){
+        if (currentTemp >= 80.0){
+        overTemperatureDetected();
+      }
+      else if(currentTemp <= 75.0){
+        safeOperatingConditions();
+      }
```

Comment:
nit: indentation is a bit off. Not a big deal since we run an auto-formatter on our actual codebase.

```suggestion
        if (currentTemp >= 80.0) {
          overTemperatureDetected();
        }
        else if(currentTemp <= 75.0) {
          safeOperatingConditions();
        }
```